src/main/java/carpet/script/value/ScreenValue.java

```diff
@@ -359,8 +361,16 @@ public String getTypeString()
 
     @Override
     public NbtElement toTag(boolean force) {
-        if (!force) throw new NBTSerializableValue.IncompatibleTypeException(this);
-        return NbtString.of(getString());
+        if(this.screenHandler == null) {
+            return new NbtList();
```

Comment:
Not sure if that's the best solution to return when the screen is null (closed)?

Reply:
Thread for example returns null tag when not finished. Why not a null nbt element here?